clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/KubernetesManifest.java

```diff
@@ -102,6 +103,32 @@ public void setNamespace(String namespace) {
   }
 
   @JsonIgnore
+  public Optional<Map<String, String>> getSpecTemplateAnnotations() {
+    if (!containsKey("spec")) {
+      return Optional.empty();
+    }
+
+    Map<String, Object> spec = (Map<String, Object>) get("spec");
+    if (!spec.containsKey("template")) {
+      return Optional.empty();
+    }
+
+    Map<String, Object> template = (Map<String, Object>) spec.get("template");
+    if (!template.containsKey("metadata")) {
+      return Optional.empty();
+    }
+
+    Map<String, Object> metadata = (Map<String, Object>) template.get("metadata");
+    Map<String, String> result = (Map<String, String>) metadata.get("annotations");
+    if (result == null) {
+      result = new HashMap<>();
+      metadata.put("annotations", result);
```

Comment:
It feels like you should do this outside of this method if it turns out `result == null`, but it took so much work to get here that it's probably not worth it.

Reply:
That's true, and it's also annoying to have your new map's edits not be automatically reflected in the manifest object.